lib/Controller/PasswordController.php

```diff
@@ -189,4 +190,30 @@ public function update($current_password, $new_password, $confirm_password, $red
 		}
 		return new RedirectResponse($redirect_url);
 	}
+
+	/**
+	 * Extracts the password requirements from app config
+	 *
+	 * @return array
+	 */
+	protected function getPasswordRequirements(): array {
+		$configKeys = [
+			'spv_min_chars',
+			'spv_lowercase',
+			'spv_uppercase',
+			'spv_numbers',
+			'spv_special_chars',
+			'spv_def_special_chars'
+		];
+
+		$passwordRequirements = [];
+
+		foreach ($configKeys as $configKey) {
+			if ($this->config->getAppValue('password_policy', $configKey.'_checked') === 'on') {
```

Comment:
taking into account the similarities among the keys, maybe it should have its own class. It could be easier to reuse.